src/formats/png.zig

```diff
@@ -63,9 +74,137 @@ pub const PNG = struct {
     }
 
     pub fn writeImage(allocator: Allocator, write_stream: *Image.Stream, image: Image, encoder_options: Image.EncoderOptions) ImageWriteError!void {
-        _ = allocator;
-        _ = write_stream;
-        _ = image;
-        _ = encoder_options;
+        const options = encoder_options.png;
```

Comment:
The `writeImage` is an adapter function. The design of zigimg is that if you don't care about the format, you use `Image`. But if you know the format you want to use, you can use the format implementation directly. 

Right now we can't write a PNG using the `PNG` struct directly.  If you look at Netpbm implementation of writing, it delegate the writing to a temporary Netbpm file filled with the information from the `Image` instance.

Reply:
I don't really understand what you mean
It seems possible to me to call `PNG.writeImage(alloc, stream, image, options)` from anywhere isn't it? What do you mean "using the PNG struct directly"?
Do you mean making adding fields to PNG that hold the metadata of a png file? (maybe there was a typo and you meant "writing to a temporary struct")

Reply:
I think I understand now by looking more at the Netbpm structure